kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/util/CtlUtils.scala

```diff
@@ -88,6 +88,50 @@ object CtlUtils {
     nodes
   }
 
+  /**
+   * List Kyuubi engine nodes info.
+   */
+  private[ctl] def listZkEngineNodes(
+      conf: KyuubiConf,
+      cliConfig: CliConfig,
+      filterHostPort: Boolean): Seq[ServiceNodeInfo] = {
+    var nodes = Seq.empty[ServiceNodeInfo]
+    withDiscoveryClient(conf) { discoveryClient =>
+      val znodeRoot = getEngineZkNamespace(conf, cliConfig)
+      val children = discoveryClient.getChildren(znodeRoot)
+      val hostPortOpt =
+        if (filterHostPort) {
+          Some((cliConfig.zkOpts.host, cliConfig.zkOpts.port.toInt))
+        } else None
+      children.map((engineId: String) => {
+        val enginePath = znodeRoot + "/" + engineId
+        nodes = getServiceNodes(discoveryClient, enginePath, hostPortOpt).++:(nodes)
+      })
+    }
+    nodes
+  }
+
+  private[ctl] def getEngineZkNamespace(conf: KyuubiConf, cliConfig: CliConfig): String = {
+    val engineType = Some(cliConfig.engineOpts.engineType)
+      .filter(_ != null).filter(_.nonEmpty)
+      .getOrElse(conf.get(ENGINE_TYPE))
+    val engineSubdomain = Some(cliConfig.engineOpts.engineSubdomain)
+      .filter(_ != null).filter(_.nonEmpty)
+      .getOrElse(conf.get(ENGINE_SHARE_LEVEL_SUBDOMAIN).getOrElse(""))
```

Comment:
this function almost is a fork of `getZkNamespace`, why not modify the original one?